lib/src/emoji_picker.dart

```diff
@@ -149,55 +156,30 @@ class EmojiPickerState extends State<EmojiPicker> {
   @override
   void initState() {
     super.initState();
-    _updateEmojiFuture = _updateEmojis();
+    _updateEmojis();
   }
 
   @override
   void didUpdateWidget(covariant EmojiPicker oldWidget) {
     if (oldWidget.config != widget.config) {
       // Config changed - rebuild EmojiPickerView completely
       _loaded = false;
-      _updateEmojiFuture = _updateEmojis();
+      _updateEmojis();
     }
     super.didUpdateWidget(oldWidget);
   }
 
   @override
   Widget build(BuildContext context) {
     if (!_loaded) {
-      // Load emojis
-      _updateEmojiFuture.then(
-        (value) => WidgetsBinding.instance.addPostFrameCallback((_) {
-          if (!mounted) return;
-          setState(() {
-            _loaded = true;
-          });
-        }),
-      );
-
-      // Show loading indicator
-      return Container(
-        alignment: Alignment.center,
-        color: widget.config.bgColor,
-        child: const CircularProgressIndicator(),
-      );
-    }
-    if (widget.config.showRecentsTab) {
-      _categoryEmoji[0].emoji = _recentEmoji.map((e) => e.emoji).toList();
+      // initialization takes less than a second in the very worst case
+      // scenario, plus it is happening usually when the widget is off-screen
+      // no need to build progress indicator
+      return const SizedBox.shrink();
```

Comment:
As mentioned in another comment above already I would feel safer to have a LoadingIndicator here or a custom Widget option. But it's okay to not address it in this PR 👍

Reply:
I don't mind adjusting this, let's discuss in that other thread.